### DISCUSSION TITLE
Handle important size files download with the Files Service

### DISCUSSION THREAD

**Discussion by ghost** [mia-platform/community#577] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

Files Service v2.10.1

### Description

Hi guys!
I've a scenario where I need to download a very heavy size file (in the GB order) from an Azure Blob Storage and I'm unsure that the Files Service can handle this kind of operations.

Here's my doubts:
- what could be the possible maximum size of a file to download with the Files Service?
- do the Files Service handles download request using streaming or multipart logic?
- is the Files Service suited for this kind of integration?

Thank you very much

**Comment by malta895** [comment, marked as answer]
Hi @ivanpasquariello,
yes, the files service uses streaming to serve downloaded files, and you shouldn't have problems in downloading files even of that big size.
Feel free to reach out in case you face any problem!
